presto-docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -23,47 +20,362 @@ connection properties as appropriate for your setup:
+
+.. warning::
+
+  Due to date and time differences in the libraries used by Presto and the
+  Oracle JDBC driver, attempting to insert or select a datetime value earlier
+  than ``1582-10-15`` results in an incorrect date inserted.
+
+.. _character mapping:
+
+Mapping character types
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto's ``VARCHAR(n)`` maps to ``VARCHAR2(n CHAR)`` if ``n`` is no greater than
+4000. A larger or unbounded ``VARCHAR`` maps to ``NCLOB``.
+
+Presto's ``CHAR(n)`` maps to ``CHAR(n CHAR)`` if ``n`` is no greater than 2000.
+A larger ``CHAR`` maps to ``NCLOB``.
+
+Using ``CREATE TABLE AS`` to create an ``NCLOB`` column from a ``CHAR`` value
+removes the trailing spaces from the initial values for the column. Inserting
+``CHAR`` values into existing ``NCLOB`` columns keeps the trailing spaces. For
+example::
+
+    presto> CREATE TABLE vals AS SELECT CAST('A' as CHAR(2001)) col;
+    presto> INSERT INTO vals (col) VALUES (CAST('BB' as CHAR(2001)));
+    presto> SELECT LENGTH(col) FROM vals;
+    2001
+    1
+
+Attempting to write a ``CHAR`` that doesn't fit in the column's actual size
+fails. This is also true for the equivalent ``VARCHAR`` types.
+
+
+Type mapping configuration properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table:: Type Mapping Properties
+  :widths: 20, 20, 50, 10
+  :header-rows: 1
+
+  * - Configuration property name
+    - Session property name
+    - Description
+    - Default
+  * - ``unsupported-type.handling-strategy``
+    - ``unsupported_type_handling_strategy``
+    - Configures how unsupported column data types are handled:
+
+      - ``IGNORE`` - column is not accessible.
+      - ``CONVERT_TO_VARCHAR`` - column is converted to unbounded ``VARCHAR``.
+
+    - ``IGNORE``
+  * - ``oracle.number.default-scale``
+    - ``number_default_scale``               
+    - Default Presto ``DECIMAL`` scale for Oracle ``NUMBER`` (without precision 
+      and scale) date type. When not set then such column is treated as not 
+      supported.
+    - not set
+  * - ``oracle.number.rounding-mode``
+    - ``number_rounding_mode``
+    - Rounding mode for the Oracle ``NUMBER`` data type. This is useful when 
+      Oracle ``NUMBER`` data type specifies higher scale than is supported in 
+      Presto. Possible values are:
+
+      - ``UNNECESSARY`` - Rounding mode to assert that the 
+        requested operation has an exact result, 
+        hence no rounding is necessary.
+      - ``CEILING`` - Rounding mode to round towards
+        positive infinity.
+      - ``FLOOR`` - Rounding mode to round towards negative
+        infinity.
+      - ``HALF_DOWN`` - Rounding mode to round towards
+        ``nearest neighbor`` unless both neighbors are
+        equidistant, in which case rounding down is used.
+      - ``HALF_EVEN`` - Rounding mode to round towards the
+        ``nearest neighbor`` unless both neighbors are
+        in which case rounding towards the nearest neighbor is
+        performed.
```

Comment:
I made suggestion fix here before. The rounding here is done differently.
```suggestion
      - ``HALF_EVEN`` - Rounding mode to round towards the
        ``nearest neighbor`` unless both neighbors are
        equidistant, in which case rounding towards the even neighbor is
        performed.
```

Reply:
ooh, I actually did fix that but looking back at the doc, there is more than one instance where similar language is used.